### PR TITLE
7.0 nl ing mt940

### PR DESCRIPTION
**Pull request by NL66278**
Update mt940 import for Dutch bank ING to latest specs.

### CODE REVIEW

File: account_banking_nl_ing_mt940/account_banking_nl_ing_mt940.py [hunk 1]
```diff
@@ -58,7 +63,9 @@ def handle_tag_60F(self, cr, data):
 
     def handle_tag_61(self, cr, data):
         super(IngMT940Parser, self).handle_tag_61(cr, data)
-        parsed_data = self.tag_61_regex.match(data).groupdict()
+        re_61 = self.tag_61_regex.match(data)
+        assert re_61, 'Cannot parse %s' % data
```
Comment (sbidoul) [inline account_banking_nl_ing_mt940/account_banking_nl_ing_mt940.py]: Since this can reveal a input data format error, I'd suggest a raise instead of assert

**Comment by sbidoul** [comment]
There are a few flake8 and pylint issues.

**Comment by NL66278** [comment]
@sbidoul Hello Stephane, I replaced the assert with a raise ValueError and fixed the flake/pylint issues.

File: account_banking_nl_ing_mt940/account_banking_nl_ing_mt940.py [hunk 2]
```diff
@@ -102,8 +113,8 @@ def handle_tag_86(self, cr, data):
             self.current_transaction.remote_owner_city = subfields['ORDP'][3]
 
         if 'REMI' in subfields:
-            self.current_transaction.message = '/'.join(
-                filter(lambda x: bool(x), subfields['REMI']))
+            self.current_transaction.message = (
+                '/'.join([x for x in subfields['REMI'] if x]))
```
Comment (hbrunn) [inline account_banking_nl_ing_mt940/account_banking_nl_ing_mt940.py]: The beauty of the expression you replaced is that it doesn't create a temporary copy of the list. You can have the same effect if you change the list comprehension to a generator expression.

TLDR: remove outermost square brackets

File: account_banking_nl_ing_mt940/account_banking_nl_ing_mt940.py [hunk 3]
```diff
@@ -57,14 +63,19 @@ def handle_tag_60F(self, cr, data):
             self.current_statement.id)
 
     def handle_tag_61(self, cr, data):
+        """Parse 61F tag containing transaction data."""
```
Comment (hbrunn) [inline account_banking_nl_ing_mt940/account_banking_nl_ing_mt940.py]: it's tag 61 (without 'F'), right?

File: account_banking_nl_ing_mt940/account_banking_nl_ing_mt940.py [hunk 4]
```diff
@@ -57,14 +63,19 @@ def handle_tag_60F(self, cr, data):
...
+        if not re_61:
+            raise ValueError(_("Cannot parse %s") % data)
+        parsed_data = re_61.groupdict()
         self.current_transaction.transferred_amount = \
             (-1 if parsed_data['sign'] == 'D' else 1) * str2float(
                 parsed_data['amount'])
         self.current_transaction.reference = parsed_data['reference']
 
     def handle_tag_86(self, cr, data):
+        """Parse 86F tag containing reference data."""
```
Comment (hbrunn) [inline account_banking_nl_ing_mt940/account_banking_nl_ing_mt940.py]: also no 'F' here?

File: account_banking_nl_ing_mt940/account_banking_nl_ing_mt940.py [hunk 5]
```diff
@@ -20,36 +21,41 @@
...
+        """allow transactions without remote account"""
         return bool(self.execution_date) and bool(self.transferred_amount)
 
 
 class IngMT940Parser(MT940, parser):
+    """Define mt940 parser for Dutch ING bank."""
     name = _('ING MT940 (structured)')
     country_code = 'NL'
     code = 'INT_MT940_STRUC'
+    footer_regex = '^-}$'
```
Comment (hbrunn) [inline account_banking_nl_ing_mt940/account_banking_nl_ing_mt940.py]: I used https://www.ing.nl/media/ING_ibp_mt940_942_format_description_the_netherlands_tcm162-45288.pdf as reference, there they state the message trailer is '-XXX' (p6) - did they release another document replacing this one?

File: account_banking_nl_ing_mt940/account_banking_nl_ing_mt940.py [hunk 6]
```diff
@@ -20,36 +21,41 @@
...
 
     tag_61_regex = re.compile(
-        r'^(?P<date>\d{6})(?P<sign>[CD])(?P<amount>\d+,\d{2})N(?P<type>.{3})'
-        r'(?P<reference>\w{1,16})')
+        r'^(?P<date>\d{6})(?P<line_date>\d{4})'
```
Comment (hbrunn) [inline account_banking_nl_ing_mt940/account_banking_nl_ing_mt940.py]: Also from the document above, I don't see a line date there. You probably used some real test file that contains this extra field, would it be an option to make it optional in the regex (`\d{0,4}`)? I doesn't seem to be used anyways

**Comment by NL66278** [comment]
Holger:
I remove the F's where they should not be, I will look into the generator / lambda thing (your first remark).
The format indeed changed since you downloaded the document mentioned. Maybe I could put in some test/options to support the old format as well.

**Comment by NL66278** [comment]
Superseded by: https://github.com/OCA/bank-payment/pull/142
